src/compiler/scala/tools/nsc/ast/DocComments.scala

```diff
@@ -543,8 +543,8 @@ trait DocComments { self: Global =>
 
       val substAliases = new TypeMap {
         def apply(tp: Type) = mapOver(tp) match {
-          case tp1 @ TypeRef(pre, sym, args) if (sym.name.length > 1 && sym.name.startChar == '$') =>
-            subst(sym, aliases, aliasExpansions) match {
+          case tp1 @ TypeRef(_, sym0, args) if (sym0.name.length > 1 && sym0.name.startChar == '$') =>
+            subst(sym0, aliases, aliasExpansions) match {
```

Comment:
note the following `sym1`; however, not very satisfactory. But the difficulty is everywhere in a generic model with trees, symbols, args, types. Occasionally, a domain-specific name is obvious.